src/main/user-impl/java/com/mysql/cj/jdbc/ha/plugins/failover/AuroraTopologyService.java

```diff
@@ -215,22 +210,36 @@ protected ClusterTopologyInfo queryForTopology(JdbcConnection conn) throws SQLEx
     long startTimeMs = System.currentTimeMillis();
 
     ClusterTopologyInfo topologyInfo = null;
-
-    try (Statement stmt = conn.createStatement()) {
-      try (ResultSet resultSet = stmt.executeQuery(RETRIEVE_TOPOLOGY_SQL)) {
-        topologyInfo = processQueryResults(resultSet);
-      }
-    } catch (SQLSyntaxErrorException e) {
-      // We may get SQLSyntaxErrorException like the following from MySQL databases:
-      // "Unknown table 'REPLICA_HOST_STATUS' in information_schema"
-      // Ignore this kind of exceptions.
+    try {
+      do {
+        try (Statement stmt = conn.createStatement()) {
+          try (ResultSet resultSet = stmt.executeQuery(RETRIEVE_TOPOLOGY_SQL)) {
+            topologyInfo = processQueryResults(resultSet);
+          }
+        } catch (SQLSyntaxErrorException e) {
+          // We may get SQLSyntaxErrorException like the following from MySQL databases:
+          // "Unknown table 'REPLICA_HOST_STATUS' in information_schema"
+          // Ignore this kind of exceptions.
+        }
+        if(initMultiWriterCluster != null && !initMultiWriterCluster
```

Comment:
Can the spacing be adjusted to match with the existing formatting?

Reply:
The purpose of this is to keep the state consistent with the boot-time mode

Reply:
If there is a mode change, it can be solved by restarting the service